docs/dev.rst

```diff
@@ -1,20 +1,522 @@
+
+The ``list_changes`` and ``retrieve_changes`` tasks detect new metadata but ignore any prior changes.
+
+To check that the snapshot was created successfully:
+
+.. code-block:: console
+
+    $ cci task run list_changes --org dev
+
+
+
+List Changes
+------------
+
+To see what components have changed in a target org:
+
+.. code-block:: console
+
+    $ cci task run list_changes --org dev
+
+.. note::
+    
+    This functionality relies on Salesforce's source tracking feature, so it's only available in scratch orgs.
+
+To include/exclude components from the list using the ``--include`` or ``--exclude`` option:
+
+.. code-block:: console
+
+    $ cci task run list_changes --org dev --include "test.*,another_regex" --exclude "something_to_exclude"
+
+The ``include`` and ``exclude`` patterns are matched against both the metadata type and name of the component.
+
+To include all changed components of specific types:
+
+.. code-block:: console
+
+    $ cci task run list_changes --org dev --types "CustomObject,CustomField"
+
+
+
+Retrieve Changes
+----------------
+
+The ``retrieve_changes`` task supports both ``sfdx`` and ``mdapi`` formatted source code. It utilizes the `SourceMember <https://developer.salesforce.com/docs/atlas.en-us.api_tooling.meta/api_tooling/tooling_api_objects_sourcemember.htm>`_ sObject to detect what has changed in an org, but also lets you be more selective regarding which components to retrieve when compared to the ``dx_pull`` task.
+
+Manual tracking of component versions also offers the possibility of retrieving some changes into one directory, and then running the task again to retrieve other changes into a different directory.
+ 
+To capture changes in an org:
+
+.. code-block:: console
+
+    $ cci task run retrieve_changes --org dev
+
+The task accepts ``include``, ``exclude``, and ``types`` options for filtering the list of changed components, and for scenarios where you don't want to retrieve all changed components.
+
+After the metadata is retrieved, the org snapshot updates so that the retrieved components are no longer included in ``list_changes``. 
+
+.. tip:: Avoid this by setting the ``snapshot`` option to False.
+
+By default, changes are retrieved into the ``src`` directory when using metadata source format, or the default ``sfdx`` package directory (``force-app``) when using ``sfdx`` source format.
+
+To retrieve into a different location using the ``--path`` option:
+
+.. code-block:: console
+
+    $ cci task run retrieve_changes --org dev --path your/unique/path
 
-Run Apex tests
+CumulusCI has multiple tasks for retrieving metadata from an org environment. See the comprehensive list of `retrieve changes`_ tasks in the cheat sheet.
+
+
+
+Push Changes
+------------
+
+Developers rarely edit code directly in an org environment, but instead use an editor or IDE like VSCode or IntelliJ. After code (or other metadata) in an editor, push these changes from your project's local repository to the target org.
```

Comment:
"After code (or other metadata) in an editor, push these changes from your project's local repository to the target org."

I'm missing a verb here, and I couldn't intuit which one you'd prefer.

Reply:
"After making edits to code or other metadata in an editor..."